### DISCUSSION TITLE
problem in 'tolerant' in latent class

### DISCUSSION THREAD

**Discussion by asghar13** [artefactory/choice-learn#189] [General]
Hi, hope you are well,
I ran ''lc_model = LatentClassSimpleMNL(n_latent_classes=3, fit_method="mle", optimizer="lbfgs", epochs=1000, tolerance=1e-20)
hist, results = lc_model.fit(elec_dataset, verbose=1)''
in your sample in https://colab.research.google.com/github/artefactory/choice-learn/blob/main/notebooks/models/latent_class_model.ipynb#scrollTo=F57NWfxAYp9b
but I got the error of 
''
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-5-b313656b728d>](https://localhost:8080/#) in <cell line: 2>()
      1 lc_model = LatentClassSimpleMNL(n_latent_classes=3, fit_method="mle", optimizer="lbfgs", epochs=1000, tolerance=1e-20)
----> 2 hist, results = lc_model.fit(elec_dataset, verbose=1)

3 frames
[/usr/local/lib/python3.10/dist-packages/choice_learn/models/simple_mnl.py](https://localhost:8080/#) in __init__(self, add_exit_choice, intercept, optimizer, lr, **kwargs)
     37             Learning Rate to be used with optimizer.
     38         """
---> 39         super().__init__(add_exit_choice=add_exit_choice, optimizer=optimizer, lr=lr, **kwargs)
     40         self.instantiated = False
     41         self.intercept = intercept

TypeError: ChoiceModel.__init__() got an unexpected keyword argument 'tolerance'
''

I know the part of '' ---> 39         super().__init__(add_exit_choice=add_exit_choice, optimizer=optimizer, lr=lr, **kwargs)
     40         self.instantiated = False
     41         self.intercept = intercept''
     is related to the SimpleMNL, but I got the similar problem for my own dataset also.
     I would be grateful for any help to solve the problem

**Comment by VincentAuriau** [comment]
Hello @asghar13 !

Thank you for trying choice-learn and reaching out.
The argument "tolerance" has been renamed into "lbfgs_tolerance" and it hasn't been well cascaded onto the Latent Class model. Thank you for pointing out this bug.

The issue should be fixed in the branch [fix-tol-lcmodel](https://github.com/artefactory/choice-learn/tree/fix-tol-lcmodel).
I will run a few tests before merging into main and pushing on PyPI to update the pip package. Are you fine working with a clone of the repository in the mean time ?

**Reply by asghar13** [reply]
Thank you for your time and addressing the issue. I really appreciate your efforts in fixing it. I can wait until the changes are merged into the main branch and the package is updated on PyPI.  

In the meantime, I will try with the `fix-tol-lcmodel` branch as suggested. Thank you again for your support.

**Comment by VincentAuriau** [comment]
Hello @asghar13,
I have release the [v1.1.1](https://github.com/artefactory/choice-learn/releases/tag/v1.1.1) that should fix the issue.
It is also avaialble on PyPI.
Let me know if you encounter any other problem. Thanks again for reporting it ! :)

**Reply by asghar13** [reply]
thanks a lot
